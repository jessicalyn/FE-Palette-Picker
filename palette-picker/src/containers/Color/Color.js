import React, { Component } from 'react';
import emptyStar from '../../assets/emptyStar.png';
import filledStar from '../../assets/filledStar.png';
import { connect } from 'react-redux';
import { toggleLocked } from '../../actions';

export default class Color extends Component {

  render() {
    const { color, locked, index } = this.props
    return(
      <div className="palette-color" style={{ backgroundColor: color }}>
        <h3>{color}</h3>
        <button onClick={this.props.handleToggle} >
          <img className="star-icon" id={index} src={emptyStar} alt={'not saved icon'}/>
        </button>
      </div>
    )
  }
}