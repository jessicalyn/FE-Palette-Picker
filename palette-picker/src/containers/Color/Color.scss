.palette-color {
  height: 100px;
  margin: 0;
  // display: flex;
  align-content: space-between
}

button {
  border: none;
  text-decoration: none;
  background: transparent;
  display: flex;
  flex-direction: column
}