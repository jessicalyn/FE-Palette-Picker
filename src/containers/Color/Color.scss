.palette-color {
  display: flex;
  height: 100px;
  margin: 0;
  justify-content: space-between;
}

button {
  border: none;
  text-decoration: none;
  background: transparent;
  display: flex;
  flex-direction: column
}

.indiv-color-text {
  font-family: $font-title;
  font-weight: bolder;
}

.star-icon {
  align-self: flex-end;
}

.indiv-color-text,
.star-icon {
  padding: 20px;
}

