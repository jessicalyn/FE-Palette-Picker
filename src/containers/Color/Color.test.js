import Color from './Color';
import React from 'react';
import { shallow } from 'enzyme';

describe('Color Container', () => {
  const mockColor = '#F1C231'
  const mockLocked = false;
  const mockIndex = 0
  const mockHandleToggle = jest.fn()

  it('should match the snapshot if locked is false', () => {
    const wrapper = shallow(
      <Color 
        color={mockColor}
        locked={mockLocked}
        index={mockIndex}
        handleToggle={mockHandleToggle}
      />
    )

    expect(wrapper.debug()).toMatchSnapshot()
  })

  it('should match the snapshot if locked is true', () => {
    const wrapper = shallow(
      <Color 
        color={mockColor}
        locked={!mockLocked}
        index={mockIndex}
        handleToggle={mockHandleToggle}
      />
    )

    expect(wrapper.debug()).toMatchSnapshot()
  })
})
