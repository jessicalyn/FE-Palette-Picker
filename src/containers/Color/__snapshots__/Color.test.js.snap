// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Container should match the snapshot if locked is false 1`] = `
"<div className=\\"palette-color\\" style={{...}}>
  <h3 className=\\"indiv-color-text\\">
    #F1C231
  </h3>
  <button onClick={[Function: mockConstructor]}>
    <img className=\\"star-icon\\" id={0} src=\\"emptyStar.png\\" alt=\\"not saved icon\\" />
  </button>
</div>"
`;

exports[`Color Container should match the snapshot if locked is true 1`] = `
"<div className=\\"palette-color\\" style={{...}}>
  <h3 className=\\"indiv-color-text\\">
    #F1C231
  </h3>
  <button onClick={[Function: mockConstructor]}>
    <img className=\\"star-icon\\" id={0} src=\\"filledStar.png\\" alt=\\"not saved icon\\" />
  </button>
</div>"
`;
